Docs/Book/RDKit_Book.rst

```diff
@@ -489,6 +490,44 @@ Primitive        Property               Notes
 <-         "dative left"         extension
 =========  ====================  ===================
 
+Hs in SMARTS
+^^^^^^^^^^^^
+
+Hs in SMARTS are interpreted as hydrogen atoms if the equivalent atom expression would also be a valid SMILES; otherwise they are interpreted as a query for any atom with a single attached hydrogen.
+
+Some examples:
+
+======  ==============
+SMARTS  Interpretation
+======  ==============  
+[H]     [#1]
+[H+]    [#1+]
+[H,Cl]  [\*H1,Cl]
+[HH]    [\*H1;\*H1]
```

Comment:
Now is that one correct? I find this edge case odd. `[HH]` is a valid smiles, not a query atom.

Reply:
Smarts interpretations and smiles interpretations can be different.

In this case the smarts can "double up" tokens like + with ands, you can make some odd patterns like:

```
>>> pat = Chem.MolFromSmarts("[HH+-+-+-H]")
>>>
```

Reply:
@bp-kelley is that still a valid SMILES? 

Should the SMILES match to itself if query is parsed as SMARTS? `[HH]` would be the only case that would be true, but for wrong reasons.

Reply:
No, from the daylight theory manual

https://www.daylight.com/dayhtml/doc/theory/theory.smarts.html

4.7 SMARTS Versus SMILES

All SMILES expressions are also valid SMARTS expressions, but the semantics changes because SMILES describes molecules whereas SMARTS describes patterns. The molecule represented by a SMILES string is usually, but not always, matched by the same string when used as a SMARTS.

Reply:
> Now is that one correct? I find this edge case odd. `[HH]` is a valid smiles, not a query atom.

But it's not of the form `[<weight>H<charge><map>]`, which are the only expressions that are interpreted as H atoms.